Fixes for refill+store coalescing

ricted98 opened the pull request:
This PR fixes few issues that arose when integrating the HPDCache@https://github.com/openhwgroup/cv-hpdcache/commit/cc95e43db4eed8be0421f232104b24aac150737b with CVA6 (see https://github.com/ricted98/cva6/commit/7f3139ab21f6b6cdabc150e4a955b2734bb7262a).


- Decouple response generation and dirty refill muxing (e.g. CVA6 does not generate responses on stores)
- Ensure correct muxing when `REFILL_REQ_RATIO=1`

cfuguet commented:
Thanks @ricted98.

Let me know when your tests with the CVA6 pass.

I'm opening a series of issues regarding the SystemC testbench to use it as a todo list, as there are many coverage holes.

cfuguet commented:
Hi @ricted98,

It looks like you are fighting against the Linter (and he is winning :)...)

If required, you can run the linter command (Verible and Verilator) with the script in rtl/lint. There is a makefile to run the corresponding commands

ricted98 commented:
@cfuguet,
I just noticed it, thanks!
I was multiplexing between students and lint fixes... sorry for the spam :smile:

cfuguet commented:
Great ! :)

I'm modifying the testbench to at least test the requests with no associated response. I propose to wait for that before merging. I expect to finish the modifications this afternoon.

cfuguet commented:
Hi @ricted98,

I've just pushed the modifications in the testbench to test posted requests (no expected response).

I also reverted your old commit implementing the refill coalescing. The modifications in the testbench allowed to trigger the issue you described.

Please rebase this branch onto the latest commit in master.

Thanks !

ricted98 commented:
Thanks for the update. I squashed and rebased the commits to include the fixed version of store+refill coalescing.
The CVA6 CI is running as well. See: https://github.com/ricted98/cva6/commit/e21dde61647d29c6b5a70d3ab887c00942adb08b
